📖Update release docs (#2802)
kubernetes-sigs/cluster-api-provider-openstack: RELEASE.md

   - Replace `vX.Y.Z` with the new release tag you're creating.
   - This command generates the release notes here
     `releasenotes/<RELEASE_TAG>.md` .

1. Next step is to clean up the release note manually.
   - If release is not an alpha or a beta or release candidate, check for duplicates,
     reverts, and incorrect classifications of PRs, and whatever release
     creation tagged to be manually checked.
   - For any superseded PRs (like same dependency uplifted multiple times, or
     commit revertion) that provide no value to the release, move them to
     Superseded section. This way the changes are acknowledged to be part of the
     release, but not overwhelming the important changes contained in the release.
   - Commit your changes, push the new branch and create a pull request:
     - The commit and PR title should be 🚀 Release v1.x.y:
       -`git commit -S -s -m ":rocket: Release vX.Y.Z"`
       -`git push -u origin release-notes-X.Y.Z`
     - Important! The commit should only contain the release notes file, nothing
     else, otherwise automation will not work.
   - Ask maintainers and release team members to review your pull request.

>    Once PR is merged following GitHub actions are triggered:

Reviewer: ```suggestion
   Once the PR is merged, the following GitHub actions are triggered:
```

   - GitHub action `Create Release` runs following jobs
     - GitHub job `push_release_tags` will create and push the tags. This action
     will also create release branch if its missing and release is `rc` or minor.
     - GitHub job `create draft release` creates draft release. Don't publish the

>      release until release tag is visible in. Running actions are visible on the

Reviewer: We need to follow image promotion below so let's just say we should not publish yet.
```suggestion
     release yet. Running actions are visible on the
```
